test/blackbox-tests/test-cases/melange/emit-with-runtime-deps.t

```diff
@@ -0,0 +1,73 @@
+Test simple interactions between melange.emit and copy_files
+
+  $ cat > dune-project <<EOF
+  > (lang dune 3.7)
+  > (using melange 0.1)
+  > EOF
+
+  $ cat > dune <<EOF
+  > (melange.emit
+  >  (alias mel)
+  >  (target output)
+  >  (runtime_deps assets/file.txt)
```

Comment:
Another thing to test would be including directory targets in this list.

Reply:
One more test case: including the same path twice.

Reply:
Maybe another test could be trying to "escape" the workspace with relative paths (not sure if this is covered already by some of the resolution functions being used)

Reply:
Added edge case tests here: [7ab3576](https://github.com/ocaml/dune/pull/7199/commits/7ab3576e0579e38e3389d1d6dec548d0fdee7059)

Added a directory target test showing it doesn't work: [99c648c](https://github.com/ocaml/dune/pull/7199/commits/99c648c6758897be4fe4a65e48f9016a0249143f)